app/controllers/devise_sessions_controller.rb

```diff
@@ -2,55 +2,80 @@ class DeviseSessionsController < Devise::SessionsController
   include ApplicationHelper
   include CookiesHelper
 
+  skip_before_action :verify_authenticity_token, only: [:create]
+
   before_action :check_login_state, only: %i[
-    create
     phone_code
-    phone_code_send
     phone_verify
     phone_resend
   ]
 
   before_action :check_password_ok, only: %i[
     phone_code
-    phone_code_send
     phone_verify
     phone_resend
   ]
 
   def create
-    render :new and return unless params.dig(:user, :password)
+    payload = get_payload(params[:jwt])
+
+    render :new and return unless params.dig(:user, :email) || params.dig(:user, :password)
+
+    @email = params.dig(:user, :email)
+    catch(:warden) do
+      self.resource = warden.authenticate(auth_options)
+    end
 
-    self.resource = warden.authenticate(auth_options)
     if resource
-      login_state.update!(password_ok: true)
+      @login_state = create_login_state(payload, @email, password_ok: true)
```

Comment:
Since the LoginState is only created after authenticating the user, we can do away with the `password_ok` thing

Reply:
That's a good point.  I'll remove this in a separate PR after this has been merged for an hour, just to make sure anyone part way through a login at the time of deployment cannot skip entering their password.

Reply:
I think we can remove references to `password_ok` in the code in this PR, since the field has a default value in the database, and then after this PR's been deployed we can deploy a migration to remove the field.